lib/engine/step/g_1882/special_nwr.rb

```diff
@@ -138,7 +138,8 @@ def available_hex(entity, hex)
             end
 
           when :place_token
-            ability(entity).hexes.include?(hex.id)
+            ability(entity).hexes.include?(hex.id) &&
+            hex.tile.cities.any? { |c| c.tokenable?(entity.owner, free: true) }
```

Comment:
so one thing to note is that avialble_hex should return an array of edges, what you have now is a boolean which is hacky and sort of works, but is not consistent with how this method is used elsewhere

Reply:
yeah, i think i ran into that previously, it seems to be subtle different between tile laying and tokening.
Maybe we should rename the tile version to available_edges which gets called by available_hex by default